unbalanced_dataset/over_sampling/smote.py

```diff
@@ -308,19 +308,19 @@ def _make_samples(self, X, y_type, nn_data, nn_num, n_samples,
         X = check_array(X)
 
         # A matrix to store the synthetic samples
-        X_new = np.zeros((n_samples, X.shape[1]))
+        X_new = np.zeros((int(n_samples), X.shape[1]))
```

Comment:
I am not a big fan of casting n_samples to int here. This is an argument that is provided by the user so let raise the warning. However, if you got a warning due to line 384 we have to do round there and cast at that point.